packages/block-library/src/gallery/edit.js

```diff
@@ -217,10 +223,18 @@ function GalleryEdit( props ) {
 			  } )
 			: selectedImages;
 
+		if ( ! imageArray.every( isValidFileType ) ) {
+			noticeOperations.removeAllNotices();
```

Comment:
I added a call to `removeAllNotices` here to be consistent with other error notices shown in the edit component. It prevents multiple notices being displayed when repeatedly dropping invalid file onto the placeholder.